src/transaction_builder.js

```diff
@@ -437,29 +448,36 @@ TransactionBuilder.prototype.__build = function (allowIncomplete) {
   return tx
 }
 
+function canSign (input) {
+  return input.hashType !== undefined &&
```

Comment:
`canSign` is probably the wrong name,  its more like `isInputReady`... but another time